tests/test_asyncio.py

```diff
@@ -72,82 +74,70 @@ async def test_recv_timeout(push_pull):
     f1 = b.recv()
     b.rcvtimeo = 1000
     f2 = b.recv_multipart()
-    with pytest.raises(zmq.Again):
+    with pytest.raises(ExceptionGroup) as excinfo:
         await f1
+    assert excinfo.group_contains(zmq.Again)
     await a.send_multipart([b"hi", b"there"])
     recvd = await f2
-    assert f2.done()
     assert recvd == [b"hi", b"there"]
 
 
 @mark.skipif(not hasattr(zmq, "SNDTIMEO"), reason="requires SNDTIMEO")
 async def test_send_timeout(socket):
     s = socket(zmq.PUSH)
     s.sndtimeo = 100
-    with pytest.raises(zmq.Again):
+    with pytest.raises(ExceptionGroup) as excinfo:
```

Comment:
Hopefully this isn't required when it's wrapped up. A single coroutine should raise a single error, right? This would be a major breaking change and a significant degradation of the API.